From 4789e7a111fed148d52f962cc9e5335d7cc57217 Mon Sep 17 00:00:00 2001
From: Madhusudhan S <madhusudhan.s@intel.com>
Date: Wed, 21 Nov 2018 16:57:56 +0530
Subject: [PATCH] usb: roles:  Add suspend/resume call-backs for Intel XHCI
 role switch driver

The role switch driver toggles the role to device mode
during initialization. Add suspend/resume call-backs to
the role switch driver. Change the mode back to host mode
before entering suspend and restore it to device mode in
the resume call-back.

Tracked-on: OAM-71949
Signed-off-by: Madhusudhan S <madhusudhan.s@intel.com>
---
 drivers/usb/roles/intel-xhci-usb-role-switch.c | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/drivers/usb/roles/intel-xhci-usb-role-switch.c b/drivers/usb/roles/intel-xhci-usb-role-switch.c
index 8c3e9f8..7a1fae4 100644
--- a/drivers/usb/roles/intel-xhci-usb-role-switch.c
+++ b/drivers/usb/roles/intel-xhci-usb-role-switch.c
@@ -195,6 +195,22 @@ static int intel_xhci_usb_remove(struct platform_device *pdev)
 	return 0;
 }
 
+static int intel_xhci_usb_suspend(struct platform_device *pdev)
+{
+        struct device *dev = &pdev->dev;
+        intel_xhci_usb_set_role(dev, USB_ROLE_HOST);
+
+        return 0;
+}
+
+static int intel_xhci_usb_resume(struct platform_device *pdev)
+{
+        struct device *dev = &pdev->dev;
+        intel_xhci_usb_set_role(dev, USB_ROLE_DEVICE);
+
+        return 0;
+}
+
 static const struct platform_device_id intel_xhci_usb_table[] = {
 	{ .name = DRV_NAME },
 	{}
@@ -208,6 +224,8 @@ static struct platform_driver intel_xhci_usb_driver = {
 	.id_table = intel_xhci_usb_table,
 	.probe = intel_xhci_usb_probe,
 	.remove = intel_xhci_usb_remove,
+	.suspend = intel_xhci_usb_suspend,
+	.resume = intel_xhci_usb_resume,
 };
 
 module_platform_driver(intel_xhci_usb_driver);
-- 
2.7.4

